clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -807,13 +807,15 @@ public InFlightRequest(RequestHeader header,
         }
 
         public ClientResponse completed(AbstractResponse response, long timeMs) {
-            return new ClientResponse(header, callback, destination, createdTimeMs, timeMs, false, response);
+            return new ClientResponse(header, callback, destination, createdTimeMs, timeMs, false, null, response);
         }
 
         public ClientResponse disconnected(long timeMs) {
-            return new ClientResponse(header, callback, destination, createdTimeMs, timeMs, true, null);
+            return new ClientResponse(header, callback, destination, createdTimeMs, timeMs, true, null, null);
         }
-
     }
 
+    public boolean discoverPeerVersions() {
```

Comment:
Also, is `discoverPeerVersions` the right name? Seems like it should be `discoverBrokerVersions` as `NetworkClient` is also used by clients.